core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -702,19 +702,19 @@ trait WhiskActionsApi extends WhiskCollectionAPI {
      * @param sequences the sequences encountered during inlining
      * @return a successful future in case the conditions were met or a failed fugure with either too many actions or recursion found
      */
-    private def inlineComponentsAndCountAtomicActions(atomicActionsCnt: Int, actionsToInline: List[String], sequences: List[String])(
+    private def inlineComponentsAndCountAtomicActions(atomicActionsCnt: Int, actionsToInline: List[FullyQualifiedEntityName], sequences: List[FullyQualifiedEntityName])(
         implicit transid: TransactionId): Future[Boolean] = {
         if (atomicActionsCnt > whiskConfig.actionSequenceLimit.toInt)
```

Comment:
do you make sure somewhere at init time that actionSequenceLimit.toInt doesn't throw an exception? WhiskConfig is not typesafe (#1093).